open_spiel/games/gin_rummy.h

```diff
@@ -35,53 +35,71 @@
 //  "knock_card"      int    set a specific knock card     (default = 10)
 //  "gin_bonus"       int    bonus for getting gin         (default = 25)
 //  "undercut_bonus"  int    bonus for an undercut         (default = 25)
+//  "num_ranks"       int    num ranks in deck             (default = 13)
```

Comment:
Please add input checks, like `num_ranks * num_suits` >= `hand_size * 2` (maybe I forgot +1 for the revealed card?) and others you can think of, that make up (implicit) invariants the code relies on and could crash (if supplied wrongly).

Reply:
Good call. I added the first check along with a few others. We no longer need the second one because I removed kMaxStockSize altogether. I realized it was only used to reserve 31 slots in the observation tensor. This was fine before the adjustable game size, but say you want to use a standard 52 card deck with 7 cards (a common recreational variant), then the max stock size goes up to 37. To handle this, we now simply reserve kDefaultNumCards (52) slots for stock size.